src/pages/participants.js

```diff
@@ -4,188 +4,388 @@ import { FaTwitter } from "react-icons/fa";
+function Request(form) {
+  function validate() {
+    // Validate first name
+    let first_name = formData.get("first_name");
+    inputErrors.first_name =
+      first_name.toLowerCase() === first_name.toUpperCase();
+
+    // Validate last name
+    let last_name = formData.get("last_name");
+    inputErrors.last_name = last_name.toLowerCase() === last_name.toUpperCase();
+
+    // Validate email
+    let email_regex = /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/;
+    inputErrors.email = !email_regex.test(formData.get("email"));
+
+    // It is compulsory to select an os
+    inputErrors.os = formData.getAll("os").length === 0;
+
+    // It is compulsory to select a level of experience
+    let experience = formData.getAll("experience");
+    inputErrors.experience = experience.length === 0;
+
+    // If the experience is not beginner, it is compulsory to say something
+    // about this experience
+    if (experience.filter(elem => elem !== "beginner").length !== 0) {
+      inputErrors.experience_notes = formData.get("experience_notes") === "";
+    } else {
+      inputErrors.experience_notes = false;
+    }
+
+    // It is required to explain the motivations for the workshop
+    inputErrors.motivations = formData.get("motivations") === "";
+
+    // It is required to read and accept the coc
+    inputErrors.coc = formData.get("coc") !== "accept";
+
+    return !Object.keys(inputErrors).some(key => inputErrors[key]);
+  }
+
+  let inputErrors = {
+    first_name: false,
+    last_name: false,
+    email: false,
+    os: false,
+    experience: false,
+    experience_notes: false,
+    motivations: false,
+    coc: false
+  };
+
+  let formData = new FormData(form);
+  let data = {
+    type: "attendee",
+    first_name: formData.get("first_name"),
+    last_name: formData.get("last_name"),
+    email: formData.get("email"),
+    legal_age: formData.get("legal_age") === "true",
+    os: formData.getAll("os"),
+    experience: formData.getAll("experience"),
+    experience_notes: formData.get("experience_notes"),
+    occupation: formData.get("occupation"),
+    motivations: formData.get("motivations"),
+    dietary: formData.get("dietary") || "None"
+  };
+
+  return {
+    data: data,
+    errors: inputErrors,
+    isValid: validate
+  };
+}
+
+class MyForm extends React.Component {
+  constructor() {
+    super();
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.state = {
+      errors: {},
+      submitted: false
+    };
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    let request = new Request(event.target);
+
+    if (request.isValid()) {
+      console.log(JSON.stringify(request.data));
+      /*fetch('/api/form-submit-url', {
+        method: 'POST',
+        body: data,
+      });*/
+      this.setState({ submitted: true });
+      event.target.reset();
+    } else {
+      this.setState({ submitted: false });
+    }
+
+    this.setState({ errors: request.errors });
+    window.scrollTo(0, 0);
```

Comment:
If memory serves, the Gatsby build process frowns upon code using `window` and `document`. Since this is not an essential feature, if it makes the build fail it can safely go away.